fpdcc/ccfp-asset-dashboard
# Add leaflet drawing for selecting assets

## Overview
For issue #97. Adds [`react-leaflet-draw`](https://github.com/alex3165/react-leaflet-draw) for selecting dynamic polygons and rectangles.

### Notes
I had trouble importing the `react-leaflet-draw` library. It seems to be an incompatibility between the latest version of react-leaflet and react-leaflet-draw. I tried using older versions of both libraries based on info I found in some Github issues but this didn't work for me. So I just imported [the `EditControl` component directly from the library's GitHub](https://github.com/alex3165/react-leaflet-draw/blob/master/src/EditControl.js) and it worked with no problems. Weird!

We have these issues for some more map UX improvements:
- #95 
- #96 
- #107 
- #108 
- #109 
- #112

Let me know if you have any other ideas for improvements.

## Testing Instructions
- Visit https://ccfp-asset-d-react-draw-crf578.herokuapp.com
- Login with the "ccfp-asset-dashboard test user" credentials in the LastPass shared folder.
- Select a project and go to the phases page.
- Select a phase.
- Click "edit assets" in the top left corner.
- Search for some assets. 
  - QA the leaflet draw toolbar.
  - Save them.
- Make sure the saved assets load.
